README.md

```diff
@@ -30,24 +34,42 @@ This team is spun off from the [Modules team](https://github.com/nodejs/modules)
 
 - [x] Implement chaining as described in the [design](doc/design/proposal-chaining-middleware.md), where the `default<hookName>` becomes `next` and references the next registered hook in the chain. https://github.com/nodejs/node/pull/42623
 
-- [ ] Move loaders off thread
+- [ ] Move loaders off thread.
 
-- [ ] Add helper/utility functions [`module`](https://nodejs.org/api/module.html) module
+   We hope that moving loaders off thread will allow us to preserve an async `resolve` hook while supporting the sync `import.meta.resolve` API. If that turns out to be unachievable, however, then:
 
-   - [ ] Start with the functions that make up the ESM resolution algorithm as defined in the [spec](https://nodejs.org/api/esm.html#resolver-algorithm-specification). Create helper functions for each of the functions defined in that psuedocode: `esmResolve`, `packageImportsResolve`, `packageResolve`, `esmFileFormat`, `packageSelfResolve`, `readPackageJson`, `packageExportsResolve`, `lookupPackageScope`, `packageTargetResolve`, `packageImportsExportsResolve`, `patternKeyCompare`. (Not necessarily all with these exact names, but corresponding to these functions from the spec.)
-   
-   - [ ] Follow up with similar helper functions that make up what happens within Node’s internal `load`. (Definitions to come.)
+   - [ ] Convert `resolve` from async to sync https://github.com/nodejs/node/pull/43363
+
+      - [ ] Add an async `resolve` to [`module`](https://nodejs.org/api/module.html) module
+
+      - [ ] Consider an API for async operations before resolution begins, such as `preImport` https://github.com/nodejs/loaders/pull/89
+
+- [ ] Provide a way to register loaders without a command-line flag, for example via a `"loaders"` field in `package.json`.
```

Comment:
It is also possible via `runMain()`: we could add another argument to accept loaders and apply them to the public instance of ESMLoader. This COULD open up `require()`-like functionality/parity.

Reply:
Sure, this bullet doesn't prescribe a method, just one possibility. If you want to suggest an addition feel free.

Reply:
I just don't want to forget about it ;)